### ISSUE TITLE
"cvsq --format CSV --without-header" outputs blank line when CSV file contains only header (no data rows).

### ISSUE THREAD

**Issue by derekmahar** [mithrandie/csvq#21]
Command `cvsq --format CSV --without-header` outputs an unnecessary blank line (newline) when the input CSV file contains only a header (no data rows).

```
$ printf "date,source_currency_code,target_currency_code,exchange_rate" |
 csvq --format CSV --without-header "SELECT * FROM STDIN"

$
```

By contrast, command `cvsq --format CSV` correctly outputs only the header:

```
$ printf "date,source_currency_code,target_currency_code,exchange_rate" |
csvq --format CSV "SELECT * FROM STDIN"
date,source_currency_code,target_currency_code,exchange_rate
$
```

This is a minor, but rather annoying bug because it means that when appending the output of `csvq --format CSV --without-header` of a CSV file that contains no rows to another CSV file, `csvq` appends a blank line.

**Comment by mithrandie** [comment]
The behavior is intended, but you are probably right.
Please wait a while, I’ll check about there is no problem if output nothing when there is no data to be output.

**Comment by ondohotola** [comment]
I use something like

	sed '/^$/d'

as a filter :-)-O

el

On 12/12/2019 09:18, Yuki wrote:
> The behavior is intended, but you are probably right.
> Please wait a while, I’ll check about there is no problem if output
> nothing when there is no data to be output.
> 

-- 
Dr. Eberhard W. Lisse   \         /      Obstetrician & Gynaecologist 
el@lisse.NA             / *      | Telephone: +264 81 124 6733 (cell)
PO Box 8421              \      /
Bachbrecht 10007, Namibia ;____/

**Comment by derekmahar** [comment]
Yes, I agree that `| sed '/^$/'` is a workaround, but users would first
need to identify `csvq --format CSV --without-header` as the source of the
blank line and then apply this filter to every instance.

**Comment by mithrandie** [comment]
Fixed and released as the version 1.11.8. Thanks.

**Comment by derekmahar** [comment]
Thank you!  The blank line is gone!

```
$ printf "date,source_currency_code,target_currency_code,exchange_rate" | csvq --format CSV --without-header "SELECT * FROM STDIN"
$
```
